kev/backends/dynamodb/db.py

```diff
@@ -0,0 +1,89 @@
+import decimal
+
+import boto3
+from boto3.dynamodb.conditions import Key
+from botocore.exceptions import ClientError
+
+from kev.backends import DocDB
+from kev.exceptions import DocNotFoundError
+
+
+class DynamoDB(DocDB):
+
+    db_class = boto3.resource
+    backend_id = 'dynamodb'
+
+    def __init__(self, **kwargs):
+        if 'aws_secret_access_key' in kwargs and 'aws_access_key_id' in kwargs:
+            boto3.Session(aws_secret_access_key=kwargs['aws_secret_access_key'],
+                aws_access_key_id=kwargs['aws_access_key_id'])
+        self._db = boto3.resource('dynamodb')
+        self.table = kwargs['table']
+        self._indexer = self._db.Table(self.table)
+
+    # CRUD Operations
+    def save(self, doc_obj):
+        doc_obj, doc = self._save(doc_obj)
+        # DynamoDB requires Decimal type instead of Float
+        for key, value in list(doc.items()):
+            if type(value) == float:
+                doc[key] = decimal.Decimal(str(value))
+        try:
+            self._indexer.put_item(Item=doc)
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                print('Table doesn\'t exist.')
+        return doc_obj
+
+    def delete(self, doc_obj):
+        self._indexer.delete_item(Key={'_id': doc_obj._doc['_id']})
+
+    def all(self, cls):
```

Comment:
On second thought lets do something like in the "5. Scans and queries do not return all data in the table." section of this blog post. https://www.uplift.agency/blog/posts/2016/03/clearcare-dynamodb